test/orange/universes/SimpleUnitTracker.test.cc

```diff
@@ -529,6 +538,26 @@ TEST_F(TwoVolumeTest, intersect)
     }
 }
 
+TEST_F(TwoVolumeTest, safety)
+{
+    SimpleUnitTracker tracker(this->params().host_ref());
+    VolumeId          outside = this->find_volume("outside");
+    VolumeId          inside  = this->find_volume("inside");
+
+    EXPECT_SOFT_EQ(1.9641016151377535, tracker.safety({2, 2, 2}, outside));
+    EXPECT_SOFT_EQ(1.3284271247461905, tracker.safety({2, 0, 2}, outside));
+    EXPECT_SOFT_EQ(0.5, tracker.safety({0, 0, 2}, outside));
+    EXPECT_SOFT_EQ(0.5, tracker.safety({0, 0, 1}, inside));
+    EXPECT_SOFT_EQ(1.5 - 1e-10, tracker.safety({1e-10, 0, 0}, inside));
+#if 0
+    // Correct result but there is a singularity at zero
+    EXPECT_SOFT_EQ(1.5, tracker.safety({0, 0, 0}, inside)); // degenerate!
+#else
+    // Actual result
+    EXPECT_SOFT_EQ(inf, tracker.safety({0, 0, 0}, inside)); // degenerate!
```

Comment:
Isn't that a "need-to-be-fixed" problem? (if so should we track it somewhere?)

Reply:
I think in practice it will be OK for two reasons. The first is that it's almost impossible to encounter this in practice (~2^{-150} if sampling on a unit length box) except if the primaries are defined at the origin (in which case they're likely high energy where MSC is less important and limited by other distance cutoffs). The second is that this would only happen if you have a single surface, in which case the safety distance is "large" anyway since we're at the center of the object, and a single step (which will be much smaller than infinity) will move to a position with the correct safety calculation.